kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h

```diff
@@ -879,6 +879,37 @@ class ResidualBasedNewtonRaphsonStrategy
         KRATOS_CATCH("");
     }
 
+    /**
+     * @brief Gets the current solution vector
+     * @param rDofSet The set of degrees of freedom (DOFs)
+     * @return A vector containing the current solution values for the given DOF set
+     * @details This method retrieves the current solution values for the provided DOF set. Each value is accessed using the equation ID associated with each DOF.
+     */
+    Vector GetCurrentSolution(DofsArrayType& rDofSet){
+        Vector this_solution(rDofSet.size());
+        for (auto& r_dof : rDofSet) {
```

Comment:
```suggestion
        for (auto& r_dof : rDofSet) {
```
Can't this be done in parallel?

Reply:
using block for each now